dashboard/src/composables/format.ts

```diff
@@ -0,0 +1,34 @@
+import moment from "moment";
+
+export function FormatDateTime(value: Date | undefined) {
+  if (!value || Number.isNaN(value.getTime())) {
+    return "";
+  }
+  return moment(value).format("YYYY-MM-DD HH:mm:ss");
+}
+
+export function FormatDateTimeString(value: string) {
+  const date = new Date(value);
+  return moment(date).format("YYYY-MM-DD HH:mm:ss");
+}
+
+export function FormatDuration(from: Date, to: Date) {
+  const diff = moment(to).diff(from);
+  return moment.duration(diff).humanize();
+}
+
+export function HumanFileSize(bytes: number, precision: number = 0): string {
+  const base = 1000;
+
+  if (bytes >= base ** 4) {
+    return `${(bytes / base ** 4).toFixed(precision)} TB`;
+  } else if (bytes >= base ** 3) {
+    return `${(bytes / base ** 3).toFixed(precision)} GB`;
+  } else if (bytes >= base ** 2) {
+    return `${(bytes / base ** 2).toFixed(precision)} MB`;
+  } else if (bytes >= base) {
+    return `${(bytes / base).toFixed(precision)} KB`;
+  } else {
+    return `${bytes} bytes`;
+  }
+}
```

Comment:
Here's perhaps a more readable alternative:

```suggestion
export function humanFileSize(bytes: number, precision = 0): string {
  const base = 1000;
  const units = ["bytes", "KB", "MB", "GB", "TB"];
  let i = 0;

  while (bytes >= base && i < units.length - 1) {
    bytes /= base;
    i++;
  }

  return `${bytes.toFixed(precision)} ${units[i]}`;
}
```